pkg/pubsub/dapr/dapr.go

```diff
@@ -0,0 +1,108 @@
+package dapr
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/dapr/go-sdk/client"
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/connection"
+)
+
+type ClientConfig struct {
+	// Enable, Disable batching per tool
+	EnableBatching bool `json:"enableBatching,omitempty"`
+
+	// Name of the component to be used for pub sub messaging
+	Component string `json:"component"`
+}
+
+// Dapr represents driver for interacting with pub sub using dapr.
+type Dapr struct {
+	// Array of clients to talk to different endpoints
+	client client.Client
+
+	// Name of the pubsub component
+	pubSubComponent string
+
+	// Enable, Disable batching per tool
+	batchingEnabled bool
```

Comment:
This is not a configurable flag. The dapr provider should signal "I do not support batching" so the pubsub system can emulate batching to compensate.

Reply:
How will users make the choice of whether they want to use batching or not? The initial idea was that they would provide this input via config. We can also choose a global flag to allow users to make that choice. But in that case, this choice will be enforced for every connection.